src/lib/add-on/clients/add-on_auto.rb

```diff
@@ -0,0 +1,328 @@
+                  Builtins.sformat(
+                    _("Make the add-on \"%1\" available via \"%2\"."),
+                    Ops.get_string(prod, "product", ""),
+                    media
+                  )
+                )
+              else
+                # just report an error
+                # TRANSLATORS: The placeholders are for the product name and the URL.
+                error_string = format(_("Failed to add product \"%s\" via\n%s."),
+                  # TRANSLATORS: a fallback string for undefined product name
+                  prod["product"] || _("<not_defined_name>"), media)
+                Report.Error(error_string)
+              end
+            elsif Ops.get_boolean(prod, "confirm_license", false)
+              accepted = AddOnProduct.AcceptedLicenseAndInfoFile( srcid )
+              if accepted == false
+                Builtins.y2warning("License not accepted, delete the repository and halt the system")
+                Pkg.SourceDelete(srcid)
+                SCR.Execute(path(".target.bash"), "/sbin/halt -f -n -p")
+              end
+            end
+
+            Ops.set(@sources, [media, pth], srcid)
+            Builtins.y2milestone("New source ID: %1", srcid)
+
+            # bugzilla #260613
+            AddOnProduct.Integrate(srcid) if srcid != -1
+
+          end while Ops.get(@sources, [media, pth], -1) == -1 &&
+            Ops.get_boolean(prod, "ask_on_error", false) == true
+          Ops.set(prod, "media", Ops.get(@sources, [media, pth], -1))
+          # Adjust "name", bnc #434708
+          if srcid != nil && srcid != -1
+            repos = Pkg.SourceEditGet
+
+            found_at = -1
+            counter = -1
+
+            Builtins.foreach(repos) do |one_repo|
+              counter = Ops.add(counter, 1)
+              if Ops.get_integer(one_repo, "SrcId", -1) == srcid
+                found_at = counter
+                raise Break
+              end
+            end
+
+            if found_at != -1
+              name = Ops.get_string(repos, [found_at, "name"], "")
+
+              # Possibility to set name in control file, bnc #433981
+              if Builtins.haskey(prod, "name")
+                name = Ops.get_string(prod, "name", "")
+                Builtins.y2milestone("Preferred name: %1", name)
+                # Or use the one returned by Pkg::RepositoryScan
+              else
+                repos_at_url = Pkg.RepositoryScan(Pkg.ExpandedUrl(media))
+                # [ ["Product Name", "Path" ] ]
+                Builtins.foreach(repos_at_url) do |one_repo|
+                  if Ops.get(one_repo, 1, "") == pth
+                    name = Ops.get(one_repo, 0, "")
+                    raise Break
+                  end
+                end
+                Builtins.y2milestone("Preferred name: %1", name)
+              end
+
+              Ops.set(repos, [found_at, "name"], name)
+              Ops.set(repos, [found_at, "priority"], Ops.get_integer(prod, "priority", -1))
```

Comment:
I'm not sure if using the `-1` default is a good idea. Pkg-bindnigs do not specify any limit for that, but the usual range is 0..200 and the default is 99.

I'd rather set the value only when it is present in the `prod` Hash instead of hardcoding any defaults here, i.e.
`Ops.set(repos, [found_at, "priority"], prod["priority"]) if prod.key?("priority")`